cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/Authorizer.java

```diff
@@ -54,7 +63,7 @@
    * @param action the {@link Action} being performed
    * @throws UnauthorizedException if the principal is not authorized to perform action on the entity
    */
-  void enforce(EntityId entity, Principal principal, Action action) throws UnauthorizedException;
+  void enforce(EntityId entity, Principal principal, Action action) throws Exception;
```

Comment:
Since this interface is exposed to extensions, we do not want to dictate the exceptions thrown from any method here. If it is an authorization error, then as the javadoc indicates, the extension must throw ``UnauthorizedException``. However, there may be additional exceptions while checking for those privileges/communicating with external systems which we don't want to unnecessarily wrap into ``RuntimeException``